hudi-common/src/main/java/org/apache/hudi/metadata/MetadataTableUtils.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.metadata;
+
+import org.apache.hudi.common.config.HoodieMetadataConfig;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.view.FileSystemViewStorageConfig;
+import org.apache.hudi.common.table.view.FileSystemViewStorageType;
+
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+
+public class MetadataTableUtils {
+
+  /**
+   * Whether to use batch lookup for listing the latest base files in metadata table.
+   * <p>
+   * Note that metadata table has to be enabled, and the storage type of the file system view
+   * cannot be EMBEDDED_KV_STORE or SPILLABLE_DISK (these two types are not integrated with
+   * metadata table, see HUDI-5612).
+   *
+   * @param metadataConfig
+   * @param fsViewStorageConfig
+   * @return {@code true} if using batch lookup; {@code false} otherwise.
+   */
+  public static boolean shouldUseBatchLookup(HoodieMetadataConfig metadataConfig,
```

Comment:
Can we not move all of this to `HoodieTableMetadataUtil` instead of creating a new class? That util class is meant for all MDT related stuff.

Reply:
both methods are actually MDT related right?

Reply:
ok i noticed `deleteMetadataTable` is already done in a slight different way in `org.apache.hudi.metadata.HoodieTableMetadataUtil`. i'll revert this refactoring and keep helpers in `HoodieTableMetadataUtil`